src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -396,12 +412,30 @@ public void UpdateTrackedDocument(Uri uri, SourceText newSourceText)
 
             // Step 5: Fork a new solution from the workspace with the LSP text applied.
             var lspSolution = workspaceCurrentSolution;
-            foreach (var (uri, workspaceDocuments) in documentsInWorkspace)
-                lspSolution = lspSolution.WithDocumentText(workspaceDocuments.Select(d => d.Id), _trackedDocuments[uri].Text);
+            // If the workspace text matched but we have source generated documents open, we can
+            // leave the normal documents as-is (the text matched) and just fork with the frozen sg docs.
+            if (!doesAllTextMatch)
+            {
+                foreach (var (uri, workspaceDocuments) in documentsInWorkspace)
+                    lspSolution = lspSolution.WithDocumentText(workspaceDocuments.Select(d => d.Id), _trackedDocuments[uri].Text);
+            }
+
+            lspSolution = lspSolution.WithFrozenSourceGeneratedDocuments(sourceGeneratedDocuments);
```

Comment:
@jasonmalinowski @CyrusNajmabadi this part is interesting - we actually end up *always* forking here because I have to pass in a date time.  I chose `DataTime.Now` above, but that means its always different, and so always forks even if the text is the same.

Is there something different I should be doing here?  Does the datetime need to be encoded in the URI somehow?

Reply:
It can't be in the URI, since that has to be stable as the document is opened over time. What is this the time used for?

Reply:
this no longer applies - I updated this so the datetime doens't matter